src/lib.rs

```diff
@@ -132,23 +140,41 @@ const OID_PE_ACME :&[u64] = &[1, 3, 6, 1, 5, 5, 7, 1, 31];
 #[allow(missing_docs)]
 /// The attribute type of a distinguished name entry
 pub enum DnType {
-	CountryName,
-	OrganizationName,
 	CommonName,
+	LocalityName,
+	StateOrProvinceName,
+	OrganizationName,
+	OrganizationalUnitName,
+	CountryName,
+	StreetName,
+	CustomDnType(Vec<u64>),
 	#[doc(hidden)]
 	_Nonexhaustive,
 }
 
 impl DnType {
 	fn to_oid(&self) -> ObjectIdentifier {
 		let sl = match self {
-			DnType::CountryName => OID_COUNTRY_NAME,
-			DnType::OrganizationName => OID_ORG_NAME,
 			DnType::CommonName => OID_COMMON_NAME,
+			DnType::LocalityName => OID_LOCALITY_NAME,
+			DnType::StateOrProvinceName => OID_STATE_OR_PROVINCE_NAME,
+			DnType::OrganizationName => OID_ORG_NAME,
+			DnType::OrganizationalUnitName => OID_ORG_UNIT_NAME,
+			DnType::CountryName => OID_COUNTRY_NAME,
+			DnType::StreetName => OID_STREET_NAME,
+			DnType::CustomDnType(ref oid) => oid.as_slice(),
 			DnType::_Nonexhaustive => unimplemented!(),
 		};
 		ObjectIdentifier::from_slice(sl)
 	}
+
+	fn from_oid(slice: &[u64]) -> Self {
+		// TODO use specific types if they exist? 
```

Comment:
It should be easy to make a match in the reverse direction. Could you do that? The compiler has smart logic to make such matches fast, at least I think I remember `nox` talking about something like that.

Reply:
Awesome!